packages/components/src/DataList/components/DataListAction/DataListAction.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import styles from "./DataListAction.css";
+import {
+  DataListActionProps,
+  DataListObject,
+  InternalDataListActionProps,
+} from "../../DataList.types";
+import { Typography } from "../../../Typography";
+import { Icon } from "../../../Icon";
+
+export function DataListAction<T extends DataListObject>(
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  _: DataListActionProps<T>,
+) {
+  return null;
+}
+
+export function InternalDataListAction<T extends DataListObject>({
+  label,
+  icon,
+  destructive,
+  onClick,
+  item,
+}: InternalDataListActionProps<T>) {
+  return (
+    <button className={styles.action} onClick={() => onClick?.(item)}>
+      {icon && <Icon name={icon} color={destructive ? "critical" : "blue"} />}
+
+      <Typography textColor={destructive ? "critical" : "blue"}>
```

Comment:
As these both are using the same logic, could be extracted into a constant

Reply:
oh, you's right! good catch!